arangod/Network/Methods.cpp

```diff
@@ -163,7 +164,13 @@ auto prepareRequest(RestVerb type, std::string path, VPackBufferUInt8 payload,
   req->header.addMeta(StaticStrings::HLCHeader,
                       arangodb::basics::HybridLogicalClock::encodeTimeStamp(timeStamp));
 
-  network::addSourceHeader(*req);
+  if (pool && pool->config().clusterInfo) {
+    auto& server = pool->config().clusterInfo->server();
+    if (server.hasFeature<AgencyFeature>() && server.isEnabled<AgencyFeature>()) {
+      auto* agent = server.getFeature<AgencyFeature>().agent();
+      network::addSourceHeader(agent, *req);
+    }
+  }
```

Comment:
This looks like a behavior change to me.
Previously, we were adding the header in case of coordinators and db servers, but now we will only add headers in case the server is an agent (i.e. the AgencyFeature is enabled).